lib/ASTGen/Sources/ASTGen/Exprs.swift

```diff
@@ -797,9 +797,43 @@ extension ASTGenVisitor {
     )
   }
 
-  func generate(macroExpansionExpr node: MacroExpansionExprSyntax) -> BridgedMacroExpansionExpr {
+  func generateObjCSelectorExpr(macroExpansionExpr node: MacroExpansionExprSyntax) -> BridgedObjCSelectorExpr {
+    fatalError("unimplemented")
+  }
+
+  func generateObjCKeyPathExpr(macroExpansionExpr node: MacroExpansionExprSyntax) -> BridgedKeyPathExpr {
+    fatalError("unimplemented")
+  }
+
+  func generate(macroExpansionExpr node: MacroExpansionExprSyntax) -> BridgedExpr {
+    // '#file', '#line' etc.
+    let magicIdentifierKind = BridgedMagicIdentifierLiteralKind(from: node.macroName.rawText.bridged)
+    if magicIdentifierKind != .none {
+      guard node.lastToken(viewMode: .sourceAccurate) == node.macroName else {
+        // TODO: Diagnose
+        fatalError("magic identifier token with arguments")
+      }
+
+      return BridgedMagicIdentifierLiteralExpr.createParsed(
+        self.ctx,
+        kind: magicIdentifierKind,
+        loc: self.generateSourceLoc(node.macroName)
+      ).asExpr
+    }
+
+    // Other built-in pound expressions.
+    switch node.macroName.rawText {
+    case "selector":
+      return self.generateObjCSelectorExpr(macroExpansionExpr: node).asExpr
+    case "keyPath":  
+      return self.generateObjCKeyPathExpr(macroExpansionExpr: node).asExpr
+    default:
+      // Fallback to MacroExpansionExpr.
```

Comment:
Also need to handle things like `#error`, `#warning`, `#assert`, `#fileLiteral`, `#colorLiteral`, and `#imageLiteral`, right? Would it be worth bridging an enum for `POUND_KEYWORD` such that we make sure we don't miss any cases here?

Reply:
Huh, I thought `#fileLiteral` et al were removed from the language, yeah we might want a bridging enum for `POUND_OBJECT_LITERAL`.

As for `#error`, `#warning`, and `#assert`, they are not expressions so we can't handle it here.
`#assert` is a statement, `#warrning` and `#error` are decl, considering how to handle them in Aston and `POUND_KEYWORD` would include some noise (i.e. `#if`, `#sourceLocation` etc.), I'm not sure `POUND_KEYWORD` bridging enum would be beneficial. I will think about it.

Anyway, thank you for reminding me!

Reply:
> As for `#error`, `#warning`, and `#assert`, they are not expressions so we can't handle it here.

We'd want to diagnose them though, right? AFAIK swift-syntax won't reject them